text/0374-jsii-ts-version.md

```diff
@@ -0,0 +1,504 @@
+# The `jsii` compiler (and `jsii-rosetta`) to follow TypeScript versioning
+
+* **Original Author(s)**: @RomainMuller
+* **Tracking Issue**: #374
+
+This RFC proposes to change the versioning scheme of the `jsii` compiler and
+`jsii-rosetta` to stop conforming to [semantic versioning][semver], and instead
+use the `major.minor` version of the [TypeScript] compiler it is built on.
+
+Other packages (such as `jsii-pacmak`, `@jsii/spec`, ...) are not affected by
+this proposal and will continue to conform to [semantic versioning][semver]. In
+particular, code generated by `jsii-pamcak` will continue to use the `1.x`
+release line of `@jsii/kernel`, `@jsii/runtime`, and the languages' runtime
+libraries, so as to remain completely backwards compatible.
+
+## Working Backwards
+
+### Release Notes entry
+
+```markdown
+Starting with this release, the `jsii` compiler and `jsii-rosetta` will no
+longer conform to [semantic versioning][semver]. Instead, its `major.minor`
+version will match that of the [TypeScript] compiler it uses (the [TypeScript]
+compiler does not conform to [semantic versioning][semver]). New features will
+only be introduced on the `jsii` release line that targets the latest
+[TypeScript] compiler.
+
+Users may need to modify their code when upgrading from one `major.minor` release
+line to another (addressing [TypeScript] language evolutions and `jsii` feature
+evolutions). While upgrading to the latest release line of `jsii` helps minimize
+the effort required for these updates, the release strategy allows developers to
+do so on their own schedule.
+
+This change is made to allow developers to benefit from the latest and greatest
+features of the [TypeScript] language without requiring the entire ecosystem to
+make the switch at the same time.
+
+Other packages in the jsii toolchain (such as `jsii-pacmak`, `@jsii/spec`, and
+others) are not affected by this change and will continue to follow [semantic
+versioning][semver] as they currently do. The bindings generated by
```

Comment:
Does this document say anything about the backwards compatibility guarantees of these tools?

For example, we could be MV-bumping on every release, and proudly claim "we are following semver", yet our customers would still hate us.

Reply:
It does in the sense that it has provisions around how you shouldn't be forced to upgrade _your_ dependency on a jsii tool because one of your dependencies or dependents has upgraded already (so every new line must be able to operate on output of previous AND future versions).